cmd/syslog/syslog.go

```diff
@@ -20,14 +23,41 @@ func main() {
 	server.SetFormat(syslog.RFC3164)
 	server.SetHandler(handler)
 
+	// check if syslog unix socket is already there then delete it.
+	if _, err := os.Stat(listenAddress); err == nil {
+		if err := os.Remove(listenAddress); err != nil {
+			log.Fatalf("error removing syslog socket %s: %s", listenAddress, err)
+		}
+	}
+
 	if err := server.ListenUnixgram(listenAddress); err != nil {
-		log.Fatal(err)
+		log.Fatalf("failed to listen to syslog unix socket %s: %s", listenAddress, err)
 	}
 
+	defer func() {
+		if err := os.Remove(listenAddress); err != nil {
+			log.Fatalf("error removing syslog socket %s: %s", listenAddress, err)
+		}
+	}()
+
 	if err := server.Boot(); err != nil {
-		log.Fatal(err)
+		log.Fatalf("failed to boot syslog server: %s", err)
 	}
 
+	// Unix sockets must be unlink()ed before being reused again.
+	// Handle common process-killing signals so we can gracefully shut down:
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
```

Comment:
what about sigkill?

Reply:
we were discussing tghat yesterday offline and it seems sigkil can't be caught or ignored hence I dropped it